lib/faraday/response/raise_error.rb

```diff
@@ -28,6 +28,8 @@ def on_complete(env)
           raise Faraday::ConflictError, response_values(env)
         when 422
           raise Faraday::UnprocessableEntityError, response_values(env)
+        when nil
+          raise Faraday::NilStatusError, response_values(env)
```

Comment:
If you make the suggestions I made above, you'll need to change this line as well 
`raise Faraday::NilStatusError, nil, response_values(env)`

Reply:
Sounds good 👍 I'll get to this later today

Reply:
So passing `nil` to this actually caused a bunch of issues for me when setting the backtrace. So I've opted to not pass it at all. Let me know if there's a better way around this.

Reply:
🤦‍♂  silly me, we're not using Faraday::NilStatusError.new so the backtrace is implicit